Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,941 @@
+            result[resultIndex++] = readCoordinate(tuples[i]);
+        }
+        return result;
+    }
+
+    function readCoordinateFromNode(node) {
+        var longitude = queryNumericAttribute(node, 'lon');
+        var latitude = queryNumericAttribute(node, 'lat');
+        var elevation = queryNumericValue(node, 'ele', namespaces.gpx);
+        return Cartesian3.fromDegrees(longitude, latitude, elevation);
+    }
+
+    var gpxNamespaces = [null, undefined, 'http://www.topografix.com/GPX/1/1'];
+    var namespaces = {
+        gpx : gpxNamespaces
+    };
+
+    function queryNumericAttribute(node, attributeName) {
+        if (!defined(node)) {
+            return undefined;
+        }
+
+        var value = node.getAttribute(attributeName);
+        if (value !== null) {
+            var result = parseFloat(value);
+            return !isNaN(result) ? result : undefined;
+        }
+        return undefined;
+    }
+
+    function queryStringAttribute(node, attributeName) {
+        if (!defined(node)) {
+            return undefined;
+        }
+        var value = node.getAttribute(attributeName);
+        return value !== null ? value : undefined;
+    }
+
+    function queryFirstNode(node, tagName, namespace) {
+        if (!defined(node)) {
+            return undefined;
+        }
+        var childNodes = node.childNodes;
+        var length = childNodes.length;
+        for (var q = 0; q < length; q++) {
+            var child = childNodes[q];
+            if (child.localName === tagName && namespace.indexOf(child.namespaceURI) !== -1) {
+                return child;
+            }
+        }
+        return undefined;
+    }
+
+    function queryNodes(node, tagName, namespace) {
+        if (!defined(node)) {
+            return undefined;
+        }
+        var result = [];
+        var childNodes = node.getElementsByTagName(tagName);
+        var length = childNodes.length;
+        for (var q = 0; q < length; q++) {
+            var child = childNodes[q];
+            if (child.localName === tagName && namespace.indexOf(child.namespaceURI) !== -1) {
+                result.push(child);
+            }
+        }
+        return result;
+    }
+
+    function queryNumericValue(node, tagName, namespace) {
+        var resultNode = queryFirstNode(node, tagName, namespace);
+        if (defined(resultNode)) {
+            var result = parseFloat(resultNode.textContent);
+            return !isNaN(result) ? result : undefined;
+        }
+        return undefined;
+    }
+
+    function queryStringValue(node, tagName, namespace) {
+        var result = queryFirstNode(node, tagName, namespace);
+        if (defined(result)) {
+            return result.textContent.trim();
+        }
+        return undefined;
+    }
+
+    function resolveHref(href, proxy, sourceUri, uriResolver) {
+        if (!defined(href)) {
+            return undefined;
+        }
+        var hrefResolved = false;
+        if (defined(uriResolver)) {
```

Comment:
`uriResolver` is a left over from the KML code you copied, it's not actually used anywhere in this file and should be removed from all of the function signatures (along with this defined block).  This also means that the `hrefResolved` bool can be removed as well.